postgres_insert_block.py

```diff
@@ -0,0 +1,62 @@
+from .postgres_base_block import PostgresBase
+
+
+class PostgresInsert(PostgresBase):
+    """A block for inserting incoming signals into a postgres database.
+
+    Properties:
+
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def process_signals(self, signals):
+        """execute an insert command for all incoming signals"""
+        for signal in signals:
+            self.execute_insert(signal.to_dict())
+        # only commit transactions when all have been attempted
+        if self.commit_all():
+            self._commit_transactions()
+
+    def execute_insert(self, data):
+        """execute an insert query for the given data"""
+        self.logger.debug('executing INSERT on data: {}'.format(data))
+        try:
+            self._cur.execute(self._build_insert_query_string(data),
+                              tuple(data.values()))
+        except:
+            self.logger.exception("Could not execute command".format())
+            self._rollback_transactions()
+
+    def _build_insert_query_string(self, data):
+        """build an INSERT SQL query based on the incoming (dictionary)
+        data. psycopg2 automatically escapes/sanitizes values passed into
+        execute and mogrify. First mogrifying the table name and data keys
+        assures escaping of those variables, and the data values will be
+        escaped when execute is called.
+        """
+        query_base = 'INSERT INTO %s (%s) VALUES (%s)'
+        query_final = self._cur.mogrify(query_base, [self.table_name(),
```

Comment:
Don't think we need `mogrify` anymore. And we'll want to run the values for `data.keys()` through the `_validate_string` method too